### DISCUSSION TITLE
Planned work on ShredOS & Nwipe

### DISCUSSION THREAD

**Discussion by PartialVolume** [PartialVolume/shredos.x86_64#219] [Announcements]
If you are thinking of requesting a new feature please browse the list of features that are already planned.

https://github.com/users/PartialVolume/projects/1/views/1

**Comment by Binary-Bear** [comment]
Hi PartialVolume

Here is my wish list for shredos.

**Default Settings:**
Please would you consider changing the default settings?

I suggest the following would be adequate for general disk wiping whilst saving unnecessary hard drive wear and users time.

menuentry "shredos" {
	linux /boot/bzImage console=tty3 loglevel=3 loadkeys=uk nwipe_options="--method=zero --verify=off --noblank --autopoweroff"
}

More experienced users who believe they require additional overwrites would still be able to manually adjust shredos to suit their requirements or for a more permanent solution edit the grub.cfg.

**Display:**
Please could there be an additional setting to the current dark option in which the text is brighter?

**Default Display:**
I understand the blue and white is traditional (DBAN) but it always reminds me of the windows blue screen of death.  Would it be possible to default to the new dark mode with brighter text as requested above or provide the user with the ability to chose the dark mode in grub.cfg?

**Final Blank Pass:**
Please could the final blank pass be automatically omitted if the user has chosen a single wipe with zeros?  It seems illogical to wipe twice with zero's.

**HPA Wipe:**
Ability to wipe the HPA.  I understand this is already planned.

**Removal of unnecessary options and code:**
Guttmann Wipe is an unnecessary wiping procedure which wastes users time and causes unnecessary hard drive wear.  I suggest this option is removed.

Mersenne Twister and ISAAC could be replaced with code from VeraCrypt for better cryptographic secure pseudo random data.  If the user requires a random overwrite then it may as well be cryptographically random.  VeraCrypt attracts a lot of attention and code scrutinisation so shredos could automatically benefit from such code inspections.

PRNG Stream option could also be removed and be replaced by the same VeraCrypt code.  The new method could perhaps be named "CSPRNG Stream".

**Plausible Deniability:**
If you adopt the request to streamline shredos and use VeraCrypt's code to provide cryptographically secure pseudo-random data you would simultaneously provide VeraCrypt users with a plausible explanation as to why their hard drives are full of cryptographically secure random data.

The CSPRNG Stream option is not to encrypt the drive as there would be no key or data to decrypt.  There is no benefit to encrypting data we intend to overwrite.  I suggest you pipe random data through VeraCrypt's encryption code, perhaps "TwoFish" for speed.  If shredos used the same code as VeraCrypt then an encrypted hard drive would be forensically indistinguishable from a shredos CSPRNG Stream wiped drive.

Mounir (the very talented developer of VeraCrypt) may assist you with any issues you might have implementing this CSPRNG Stream feature as he is very generous with his time and skills.

**Hot Swap:**
The ability to be able to hot swap hard drives without having to reboot shredos would be wonderful.

**Two Stage Boot**
Enable users with vintage main-boards to boot shredos much faster.  I have to declare a personal interest in this one!

Thank you.
